src/components/language/ethnologue-language/ethnologue-language.service.ts

```diff
@@ -149,32 +150,19 @@ export class EthnologueLanguageService {
 
     // filter out all of the undefined values so we only have a mapping of entries that the user wanted to edit
     const valueSetCleaned = pickBy(valueSet, (v) => v !== undefined);
-    const valueSetCleanedKeys = Object.keys(valueSetCleaned);
 
-    for (const key of valueSetCleanedKeys) {
-      const q = await this.db
-        .query()
-        .match([
-          node('user', 'User', { id: session.userId }),
-          relation('in', 'memberOfSecurityGroup', 'member'),
-          node('security', 'SecurityGroup'),
-          relation('out', 'sgPerms', 'permission'),
-          node('perm', 'Permission', {
-            property: `${key.replace('.value', '')}`,
-            edit: true,
-          }),
-          relation('out', 'permsOfBaseNode', 'baseNode'),
-          node('ethnologueLanguage', 'EthnologueLanguage', { id: id }),
-        ])
-        .return(['user', 'perm', 'ethnologueLanguage'])
-        .first();
-      if (!q) {
-        throw new UnauthorizedException(
-          `You do not have permission to edit his object', '${key}`
-        );
-      }
-    }
+    // even though we already have a cleaned value set, we still need to get them in a readable way for the verifyCanEditChanges function
+    const realChanges = await this.db.getActualChanges(
+      { ...ethnologueLanguage, createdAt: CalendarDate.now() }, // hacking in a createAt field, because getActualChanges is expecting a Resource shape.
```

Comment:
I think this would not be needed with the change I suggested to getActualChanges